hooks/gate-scripts/lib/commit_scope.py

```
@@ -0,0 +1,551 @@
+    attempted. An earlier revision armed the alarm only around the read and left
+    exactly that window unbounded; a hang there is a bypass, because the 10s
+    PreToolUse timeout emits NO decision and the harness reads that as allow.
+
+    Signals are Unix-only and main-thread-only. Where SIGALRM cannot be armed
+    this degrades to unbounded, and the surrounding hook timeout is the only
+    backstop — accepted because Claude Code hooks run on Unix, and the
+    alternative (thread-based bounding) is more machinery than the residual
+    warrants.
+    """
+
+    def __enter__(self):
```

Ruff (0.16.1)

[warning] Missing return type annotation for special method `__enter__`

(ANN204)